### PR TITLE
Cluster config

### PR DESCRIPTION
**Pull request by afhuertass**
Added text input for the threshold value as well as an apply button that needs to be pressed in order to launch the cluster computation.
I also added the loading component so now when the clusters are being computed the user can see that something is going on, and also modified some names and colors here and there.

### CODE REVIEW

File: components/cluster.py [hunk 1]
```diff
@@ -116,74 +118,67 @@ def build_cluster_component(mindate, maxdate, session_id, idd="1"):
     id_component = "date-pick-{}".format(idd)
     date_component = date_picker.get_component(
         mindate, maxdate, None, id_component)
+    threshold_component = cluster_config.get_component(idd)
     return html.Div([
         date_component,
+        html.Div(children="Type threshold:"),
```
Comment (llmlks) [inline components/cluster.py]: These might make more sense if they were added to the `cluster_config` component. Adding the class `config-label` to the "Type threshold" label will keep its styling consistent with the rest of the views

File: components/cluster.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
 import dash_html_components as html
 import dash_core_components as dcc
 from dash.dependencies import Input, Output, State
+from dash.exceptions import PreventUpdate
```
Comment (llmlks) [inline components/cluster.py]: This isn't used, as a few of the other imports, too. Maybe go through them and remove the unused ones

File: components/config/cluster_config.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import dash_html_components as html
+import dash_bootstrap_components as dbc
+import dash_core_components as dcc
+
+
+def get_component(id="1"):
+
```
Comment (llmlks) [inline components/config/cluster_config.py:11]: Function docstring missing

File: components/config/date_picker.py [hunk 4]
```diff
@@ -19,7 +19,7 @@ def get_component(min_date, max_date, default_end_date, id="date-pick"):
         end_date = default_end_date.date()
 
     return html.Div(children=[
-        html.Div(className='config-label', children='Time range'),
+        html.Div(className='config-label', children='Select dates'),
```
Comment (llmlks) [inline components/config/date_picker.py:22]: A good change, could you change the corresponding values in the instructions, too?

**Review by llmlks — CHANGES_REQUESTED** [review]
Functionality works great, a few minor adjustments. Could you also update the clustering instructions in `instructions.md` so they reflect the current features?

File: components/config/cluster_config.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    idd = "thr_{}".format(id)
+    return html.Div([
+        dcc.Input(id=idd, type="number",  value=1e-5),
```
Comment (llmlks) [inline components/config/cluster_config.py]: The other configs use `dbc.Input`, so it might be a good idea here as well

**Comment by afhuertass** [comment]
There is one issue I can't figure out, the size of the input box for the threshold is larger than expected, any idea how to control this? I tried wrapping it around a Div and I thought it should be the size of the date component ( that is what I see from how the component is built ) how can I fix it?

**Comment by llmlks** [comment]
You could add the picker's ID to the list of element IDs in the `config.css` where the element width is defined as 288px

**Comment by llmlks** [comment]
And at the same time, you could add the `Apply` buttons' IDs to the list with `#apply` to ensure there is some margin between the elements

**Review by llmlks — APPROVED** [review]
Happy with the changes. The branch is ready for merge once you've resolved the merge conflicts,

**Comment by llmlks** [comment]
Is there a reason you deleted the `opacity-unit`, `opacity-value` and `template-id` stylings from `config.css` during merge? :D

**Comment by llmlks** [comment]
Here's how they used to be: https://github.com/llmlks/QuakeWatch/pull/37/commits/e668016e243d736e608c6ee1e925544d420b4540

**Comment by afhuertass** [comment]
ah damn that was a mistake, sorry  . how can I fix it :s

**Comment by llmlks** [comment]
No worries! You can just commit the original values to this branch and make a new pull request

**Pull request merged**